### PR TITLE
Change the default stance to Defend for human players

### LINKED ISSUE #3285: Change default unit stance to defend for land units and returnfire for cloaked units [$48]
Nobody actually wants them to default to AttackAnything.

<bountysource-plugin>
---
There is a **[$48 open bounty](https://www.bountysource.com/issues/335981-change-default-unit-stance-to-defend-for-land-units-and-returnfire-for-cloaked-units?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by obrakmann**
Exceptions are dogs and stealthy things.

Closes #3285.

Thanks to @matija-hustic for laying the groundwork that makes this work properly in the first place!

### CODE REVIEW

File: mods/cnc/rules/vehicles.yaml [hunk 1]
```diff
@@ -151,6 +151,7 @@ ARTY:
 	WithMuzzleOverlay:
 	AutoTarget:
 		InitialStance: Defend
```
Comment (abcdefg30) [inline mods/cnc/rules/vehicles.yaml]: This can be removed then?

File: mods/cnc/rules/vehicles.yaml [hunk 2]
```diff
@@ -491,6 +492,7 @@ MLRS:
 		AmmoPoolName: primary
 	AutoTarget:
 		InitialStance: Defend
```
Comment (abcdefg30) [inline mods/cnc/rules/vehicles.yaml]: Same here.

**Comment by pchote** [comment]
I'm still very wary about this, but we can always revert it if (when?) our non-pro players complain / file confused bug reports about broken unit AI.

**Comment by obrakmann** [comment]
Updated. The date in the upgrade rules will probably need to be changed before merging.

**Comment by pchote** [comment]
Adding to the milestone because if we don't take this before the first playtest then I will insist that we leave it to the next cycle.  Slipping this in later in the testing cycle is asking for trouble.

**Comment by obrakmann** [comment]
Rebased onto bleed, also dropped the commit that set the `Playable` flag on the scripted players to `false`, since that's already the default.

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 3]
```diff
@@ -24,8 +24,11 @@ public class AutoTargetInfo : ITraitInfo, Requires<AttackBaseInfo>
 		[Desc("Set to a value >1 to override weapons maximum range for this.")]
 		public readonly int ScanRadius = -1;
 
-		[Desc("Possible values are HoldFire, ReturnFire, Defend and AttackAnything.")]
-		public readonly UnitStance InitialStance = UnitStance.AttackAnything;
+		[Desc("Possible values are HoldFire, ReturnFire, Defend and AttackAnything. Used for computer-controlled players.")]
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: Could you clarify this a bit wrt mission players?

**Comment by pchote** [comment]
Code changes look good, but I haven't tested this so :+0.5:.

File: OpenRA.Mods.Common/Traits/AutoTarget.cs (hunk 3 above)
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: Hmm, like this? Not sure what kind of clarification you're asking for. I updated the `Desc` and renamed the variable to `InitialStanceAI`.

Comment (pchote) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: That's better, thanks.

**Comment by penev92** [comment]
You missed fremen with the "stealthy things".

Other than that :+1:

**Comment by obrakmann** [comment]
Added default stance for fremen. Also changed the stances of all stealthy units to "HoldFire" for human players and "ReturnFire" for the AI, as discussed in IRC.

File: mods/ra/maps/fort-lonestar/map.yaml [hunk 4]
```diff
@@ -654,7 +654,7 @@ Rules:
 		Health:
 			HP: 200
 		AutoTarget:
-			InitialStance: Defend
+			InitialBotStance: Defend
```
Comment (Mailaender) [inline mods/ra/maps/fort-lonestar/map.yaml]: I guess this has to be `InitialStanceAI` to match the code although `InitialBotStance` has better readability.

Comment (obrakmann) [inline mods/ra/maps/fort-lonestar/map.yaml]: Well spotted! I only ran sed on the rules directory and forgot about this.

**Comment by abcdefg30** [comment]
Looks good to me, too. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9de9115f976ea6a64baeb94602ddde442fd785e7...8a09b406e165173328c7edc4c0f60de77828cc4a)

**Pull request merged**